### DISCUSSION TITLE
Welcome to krossbow Discussions!

### DISCUSSION THREAD

**Discussion by joffrey-bion** [joffrey-bion/krossbow#90] [General]
## 👋 Welcome!

We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by hatteajay1212** [comment]
Hello @joffrey-bion,
I'm having difficulty connecting to two URLs simultaneously. While I've successfully subscribed to a topic through the first connection and am receiving messages, I'm struggling to establish a connection or subscribe to the second URL.I have implemented stomp connections as below : 

for e.g.
val wsClient = KtorWebSocketClient(httpClient)
val client = StompClient(wsClient) {
                connectionTimeout = 60.seconds
                heartBeat = HeartBeat(minSendPeriod = 10.seconds)
                headers {
                    mapOf("username" to "H4000")
                }
}

val session = client.connect("wss://exampleurl.com/some-service/some-endpoint")
val subscription = session.subscribeText("/topic/some-topic")

 val job = GlobalScope.launch {
                subscription.collect{message->
                    println("Received Message -> $message")
               }
 }

I have connected to second url similarly.When app is launched, it crashed after few seconds with following error :

Error - org.hildan.krossbow.stomp.WebSocketClosedUnexpectedly: the WebSocket was closed while subscriptions were still active. Code: 1006 Reason: Connection was closed without close frame

I'm wondering if there are any life cycle methods available for managing Stomp connections. It's proven challenging to ascertain whether a connection has been established or not. The same issue arises when dealing with topic subscriptions and unsubscriptions.

If there are no existing provisions for life cycle methods, do you have plans to release them in the near future? I would greatly appreciate your feedback on this matter.

**Reply by joffrey-bion** [reply]
Hi @hatteajay1212, thanks for reaching out. Could you please clarify a few points?

1. Do you reproduce the problem consistently? Does it crash everytime with some reproducer code? If that's the case, it would be amazing if you could share a piece of code that reproduces the issue.
2. Are you using the same `StompClient` for both STOMP connections? If not, are the 2 stomp clients using the same web socket client? If not, are the 2 web socket clients using the same `HttpClient`?
3. If you are sharing some client at some level, could you please try without sharing anything, with independent clients at every level? Do you reproduce the problem when doing so?

> I'm wondering if there are any life cycle methods available for managing Stomp connections

What exactly are you looking for? There is an instrumentation mechanism that you can use by [setting a `KrossbowInstrumentation` in the config](https://joffrey-bion.github.io/krossbow/kdoc/krossbow-stomp-core/org.hildan.krossbow.stomp.config/-stomp-config/instrumentation.html), so you can hook on different lifecycle events such as web socket closure/error. Would that solve your problem?

If not, could you please describe what you would like to do with the information about the connection status, and how it would be best to get it delivered?
